src/agent/db_manager/src/sqlite_manager.cpp

```diff
@@ -0,0 +1,372 @@
+
+    Execute(queryString);
+}
+
+void SQLiteManager::Execute(const std::string& query)
+{
+    try
+    {
+        std::lock_guard<std::mutex> lock(m_mutex);
+        m_db->exec(query);
+    }
+    catch (const std::exception& e)
+    {
+        LogError("Error during database operation: {}.", e.what());
+        throw;
+    }
+}
+
+std::vector<Row> SQLiteManager::Select(const std::string& tableName,
+                                       const Names& fields,
+                                       const Criteria& selCriteria,
+                                       LogicalOperator logOp,
+                                       const Names& orderBy,
+                                       OrderType orderType,
+                                       unsigned int limit)
+{
+    std::string selectedFields;
+    if (fields.empty())
+    {
+        selectedFields = "*";
+    }
+    else
+    {
+        std::vector<std::string> fieldNames;
+        fieldNames.reserve(fields.size());
+
+        for (const auto& col : fields)
+        {
+            fieldNames.push_back(col.Name);
+        }
+        selectedFields = fmt::format("{}", fmt::join(fieldNames, ", "));
+    }
+
+    std::string condition;
+    if (!selCriteria.empty())
+    {
+        std::vector<std::string> conditions;
+        for (const auto& col : selCriteria)
+        {
+            if (col.Type == ColumnType::TEXT)
+                conditions.push_back(fmt::format("{}='{}'", col.Name, col.Value));
+            else
+                conditions.push_back(fmt::format("{}={}", col.Name, col.Value));
+        }
+        condition = fmt::format("WHERE {}", fmt::join(conditions, fmt::format(" {} ", MAP_LOGOP_STRING.at(logOp))));
+    }
+
+    if (!orderBy.empty())
+    {
+        std::vector<std::string> orderFields;
+        orderFields.reserve(orderBy.size());
+        for (const auto& col : orderBy)
+        {
+            orderFields.push_back(col.Name);
+        }
+        condition += fmt::format(" ORDER BY {}", fmt::join(orderFields, ", "));
+        condition += fmt::format(" {}", MAP_ORDER_STRING.at(orderType));
+    }
+
+    if (limit > 0)
+    {
+        condition += fmt::format(" LIMIT {}", limit);
+    }
+
+    std::string queryString = fmt::format("SELECT {} FROM {} {}", selectedFields, tableName, condition);
+
+    std::vector<Row> results;
+    try
+    {
+        std::lock_guard<std::mutex> lock(m_mutex);
+        SQLite::Statement query(*m_db, queryString);
+
+        while (query.executeStep())
+        {
+            int nColumns = query.getColumnCount();
+            Row queryFields;
+            queryFields.reserve(static_cast<size_t>(nColumns));
+            for (int i = 0; i < nColumns; i++)
+            {
+                queryFields.emplace_back(query.getColumn(i).getName(),
+                                         ColumnTypeFromSQLiteType(query.getColumn(i).getType()),
+                                         query.getColumn(i).getString());
```

Comment:
What if the value is not a string?

Reply:
Sqlitecpp returns the representation of the value as a string, which can then be converted to any type, so this is not an issue.